pkg/api/daemon.go

```diff
@@ -53,10 +56,31 @@ func Run(context *clusterd.Context, c *Config) error {
 		return fmt.Errorf("failed to write connection config. %+v", err)
 	}
 
+	CephMonConfWatcher(context, c)
 	ServeRoutes(context, c)
 	return nil
 }
 
+func CephMonConfWatcher(context *clusterd.Context, c *Config) {
+	go func() {
+		w, err := context.Clientset.CoreV1().ConfigMaps(c.namespace).Watch(metav1.ListOptions{})
```

Comment:
do we need a filter on the name of the configmap? `rook-ceph-mon-endpoints`

Reply:
Yes. This is how to do this.
```
opts := metav1.ListOptions{
  FieldSelector: fields.OneTermEqualSelector(api.ObjectNameField, "rook-ceph-mon-endpoints").String(),
}
kubeClient.Core().ConfigMaps(c.namespace).Watch(opts)
```

This way if a different configmap is updated, this would not be triggerred :)

Reply:
@kokhang Thank you for this! I didn't know the `FieldSelector` was used for this. :+1: